src/middleware/passport/strategies/jwt/jwt-strategy.ts

```diff
@@ -0,0 +1,46 @@
+import passport from 'passport';
+import { Request, Response, NextFunction } from 'express';
+import { isTokenBlacklisted, verifyJWT } from './jwt-utils';
+
+/**
+ * JWT authentication middleware for Express.
+ *
+ * This middleware uses Passport to authenticate the JWT token. It performs
+ * several checks to ensure the token is valid, not blacklisted, and associated
+ * with a valid user. If the checks pass, the user information is attached to
+ * the request object, and the request is passed to the next middleware or route
+ * handler. If any check fails, a 401 Unauthorized response is returned.
+ *
+ * @param {Request} req - The Express request object.
+ * @param {Response} res - The Express response object.
+ * @param {NextFunction} next - The next middleware function in the Express stack.
+ */
+export const authenticateJwt = (req: Request, res: Response, next: NextFunction) => {
+    passport.authenticate('jwt', { session: false }, async (err: any, user: any) => {
```

Comment:
How come this is not wrapped in try catch? you should be consistent :D

Reply:
cause passport.authenticate('jwt', ()=> {}) provided by npm package and to have error there i think unexpected

Reply:
still, try triggering some errors, and se if it will crash the app.

Reply:
Please show me how try trigger some errors and I will change :)

Reply:
try sending wrong options

Reply:
what You mean wrong options?
Errors are catching in callback err parameter